pinecone/client.go

```diff
@@ -273,6 +273,104 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
 	return nil
 }
 
+// ConfigureIndex is used to [scale a pods-based index] up or down by changing the size of the pods or the number of
+//replicas.
+//
+// Parameters:
+//  - name: The name of the index to configure.
+//  - pods: (Optional) The pod size to scale the index to (e.g. for a "p1" pod type,
+//  you could pass "p1.x2" to scale your index to the "x2" size,
+//  or you could pass "p1.x4" to scale your index to the "x4" size, and
+//  so forth.
+//  - replicas: (Optional) The number of replicas to scale the index to.
+//   This is capped by the maximum number of replicas allowed in your Pinecone project. To configure this number,
+//   go to [app.pinecone.io], select your project, and configure the maximum number of pods.
+//
+// Note: You can only scale an index up, not down. If you want to scale an index down,
+// you must create a new index with the desired configuration.
+//
+// Returns a ConfigureIndexResponse object, which contains the new configuration of the index, or an error.
+//
+// Example for a pods-based index originally configured with 1 "p1" pod of size "x2" and 1 replica:
+//  // To scale the size of your pods from "x2" to "x4":
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", nil)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", nil, 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale both the size of your pods and the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+// [scale a pods-based index]: https://docs.pinecone.io/guides/indexes/configure-pod-based-indexes
+// [app.pinecone.io]: https://app.pinecone.io
+func (c *Client) ConfigureIndex(ctx context.Context, name string, pods *string,
+	replicas *int32) (*control.ConfigureIndexResponse,
+	error) {
+
+	var podType *control.PodSpecPodType
+	var replicasAmt *control.PodSpecReplicas
+
+	if pods == nil && replicas == nil {
+		return nil, fmt.Errorf("Must specify either pods or replicas")
+	}
+
+	if pods != nil {
+		podTypeVal := control.PodSpecPodType(*pods)
+		podType = &podTypeVal
+	}
+	if replicas != nil {
+		replicasVal := control.PodSpecReplicas(*replicas)
+		replicasAmt = &replicasVal
+	}
+
+	request := control.ConfigureIndexRequest{
+		Spec: struct {
+			Pod struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			} `json:"pod"`
+		}{
+			Pod: struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			}{
+				PodType:  podType,
+				Replicas: replicasAmt,
+			},
+		},
+	}
+
+	req, err := c.restClient.ConfigureIndex(ctx, name, request)
```

Comment:
Can we rename `req` to `res` to make it clearer it's the `http.Response` from the function call here?

Reply:
Sure!